File: endpoint/templates/go/echo/endpoint.controller.go
package <%= nameLowerCase %>controller

import (
	_ "encoding/json"
	_ "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/<%= feature %>/dao"
	_ <%= nameLowerCase %> "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/<%= feature %>/model"
	"gopkg.in/labstack/echo.v2"
	_ "io/ioutil"
	_ "net/http"
)

func GetAll(c echo.Context) {
}

func GetById(c echo.Context) {
}

func New(c echo.Context) {
}

func Update(c echo.Context) {
}

func Remove(c echo.Context) {
}

File: endpoint/templates/node/koa/babel/endpoint.controller.js
import <%= name %>DAO from '../dao/<%= name %>-dao';

export default class <%= name %>Controller {
  async getAll() {
    try {
      let <%= nameLowerCase %>s = await <%= name %>DAO.getAll();
      this.body = <%= nameLowerCase %>s;
      this.status = 200;
    } catch(e) {
      this.status = 400;
    }
  }

  async createNew() {
    let _<%= nameLowerCase %> = this.body;

    try {
      this.body = await <%= name %>DAO.createNew(_<%= nameLowerCase %>);
      this.status = 201;
    } catch(e) {
      this.status = 400;
    }
  }

  async removeById() {
    let _id = this.params.id;

    try {
      await <%= name %>DAO.removeById(_id);
      this.status = 200;
    } catch(e) {
      this.status = 400;
    }
  }
}
